canvas.py

```diff
@@ -0,0 +1,72 @@
+import http.client
+import json
+import logging
+import time
+
+__author__ = 'Islam Elnabarawy'
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+class CanvasAPI:
+    REQUEST_HEADER = {
+        "Authorization": "Bearer ",
+        "Content-Type": "application/json"
+    }
+    WEBSITE_ROOT = "https://mst.instructure.com"
+
+    def __init__(self, canvas_token):
+        self.REQUEST_HEADER['Authorization'] += canvas_token
+
+    def _request(self, method: str, url: str, params: dict = None, retries: int = 3) -> http.client.HTTPResponse:
```

Comment:
Woah, this is pretty neat. How long has python had optional typing for functions?